lib/msf/core/module/external.rb

```diff
@@ -152,6 +152,11 @@ def handle_credential_login(data, mod)
     credential_data[:private_type] = :password
   end
 
+  if data.has_key?(:domain)
```

Comment:
My apologies, this will also need to be changed since all keys in `data` are strings:
```suggestion
  if data.has_key?('domain')
```

Reply:
So this isn't working either, (though it could be an issue). When I remove the conditional completely, and just blindly add it to the credential_data, it still isn't appearing in the creds database as a realm. I've tried this multiple ways, including:
- `  credential_data.merge({realm: data['domain']})`
- `  credential_data[:realm] = data['domain']`
So I'm not sure why it isn't getting added at this point 😕

Reply:
So I have verified in external.rb that it is added to the credential_data prior to create_credential being called. After it is called however, the realm doesn't appear in login_data so it isn't sticking in the create_credential call. I'm trying to track down which create_credential is being called and why it isn't sticking the :realm

Reply:
@acammack-r7 I got it! It turns out the fix was using :realm_key and :realm_value. I then added this is to password reporting, since it didn't exist there either. This should be good to go now!